### PR TITLE
[AD-352] Write documentation about configuration

### PR DESCRIPTION
**Pull request by gromakovsky**
**YT issue:** https://issues.serokell.io/issue/AD-352

**Checklist:**

- [x] Updated docs if necessary
  - [x] [README](README.md)
  - [x] [TUI usage guide](docs/usage-tui.md)
- [x] Adressed HLint warnings and hints
- [x] Rebased branch on current master and squashed all "Address PR comment" commits
- [x] Tested my changes if they modify code

**Description:**

I've expanded documentation of configuration and moved it into a separate file (from README). Also updated pull request template.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+* `@DATA` is a system-dependent standard directory for local
+data (`~/.local/share` on Linux or MacOS, `%APPDATA%` on
+Windows) suffixed with `/ariadne`.
+* `@PWD` is the directory from which Ariadne executable is launched.
+
+If configuration file is not found, default configuration will be used
+which is compatible with Cardano Mainnet.
+
+Each option from the configuration can be overridden from the command
+like. Use `--help` flag for more details.
```
Comment (maksbotan) [inline docs/configuration.md]: s/like/line/

File: docs/configuration.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+components: Cardano, Wallet, Update, History. In the following we
+describe configuration of each component in detail.
+
+## Cardano configuration
+
+Cardano configuration contains parameters which are used by the
+underlying Cardano node.
+
+`configuration-options` defines `Configuration` used by Cardano
+node. For more details on that please read the `docs/configuration.md`
```
Comment (maksbotan) [inline docs/configuration.md]: Maybe add a link?

File: docs/configuration.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+file in the `cardano-sl` repository. If `configuration-path` doesn't
+exist, static configuration will be dumped to that path.
+
+`log-config` is the path to the logging configuration (as used by
+`log-warper`). If it's not provided a reasonable default configuration
+will be used.
+
+`log-prefix` is the file prefixed to all paths in logging
```
Comment (maksbotan) [inline docs/configuration.md]: Maybe s/file/string?

File: docs/configuration.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+`version-check-url` is used to get latest software version.
+
+`update-url` is displayed in the message about new version.
+
+`check-delay` specifies (in seconds) how often we check for a new version.
+
+## History configuration
+
+History configuration has only one value with the key `path`. It's a
```
Comment (maksbotan) [inline docs/configuration.md]: Default?

Comment (gromakovsky) [reply docs/configuration.md]: Default what?

Comment (maksbotan) [reply docs/configuration.md]: Path. Not important, probably

File: docs/configuration.md (hunk 3 above)
Comment (gromakovsky) [reply docs/configuration.md]: I'd say s/file/path. Because it's prefixed as a path, with separator.

Comment (gromakovsky) [reply docs/configuration.md]: Also probably s/prefixed/prepended.

**Pull request merged**